openai-assistant/pom.xml

```diff
@@ -8,7 +8,7 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <java.compiler.compliance>17</java.compiler.compliance>
+    <java.compiler.compliance>21</java.compiler.compliance>
```

Comment:
17 would server both platforms.

Reply:
the fact that we run and compile with java21 does not enforce us to comply with that language lever. newer compiler can still generate java17 compilation units.